prototypes/emdpg_waveguide/EM_waveguide.cc

```diff
@@ -0,0 +1,2451 @@
+/**
+ * @brief Necessary includes for the implementation of the DPG method for the
+ * time-harmonic maxwell equations.
+ *
+ * @author Oreste Marquis, Bruno Blais, Matthias Maier, 2025
+ */
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/convergence_table.h>
+#include <deal.II/base/function.h>
+#include <deal.II/base/index_set.h>
+#include <deal.II/base/parameter_acceptor.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/timer.h>
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/distributed/grid_refinement.h>
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_nedelec_sz.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/generic_linear_algebra.h>
+#include <deal.II/lac/lapack_full_matrix.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/sparse_direct.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/sparsity_tools.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/data_out_faces.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <complex>
+#include <fstream>
+#include <iostream>
+
+/**
+ * @namespace LA
+ * @brief Alias namespace for dealii::LinearAlgebraTrilinos.
+ *
+ * This namespace provides a convenient shorthand for accessing the
+ * Trilinos-based linear algebra functionality from the deal.II library.
+ * All classes and functions from dealii::LinearAlgebraTrilinos can be
+ * accessed via the LA namespace.
+ */
+namespace LA
+{
+  using namespace dealii::LinearAlgebraTrilinos;
+} // namespace LA
+namespace EM_DPG
+{
+  using namespace dealii;
+
+  // This is a helper function that maps a 3D tensor to a 2D face in
+  // H^-1/2(curl).
```

Comment:
I don't really mind for a prototype, but if you want you, yon can add the doxygen briefs